In [1]:
import pandas as pd 
import altair as alt
import scipy.stats
import matplotlib.pyplot as plt
import pingouin as pg

In [2]:
data = pd.read_csv("SurveyData.csv")

In [3]:
data.head()

Experimenter  PID  Interface  \
0         Team    1          2   
1         Team    2          2   
2         Team    3          2   
3         Team    4          2   
4         Team    5          2   

   Rank Placement of Create and Marketplace Button Based on Where Expected  \
0                                                  2                         
1                                                  3                         
2                                                  1                         
3                                                  2                         
4                                                  1                         

  Often Intend to Click Certain Button and Get Unexpected Results  \
0                                                 No                
1                                                Yes                
2                                                Yes                
3                                                 No                
4                                                 No                

   Rank Use on Instagram in Terms of Frequency - Explore Page  \
0                                                  2            
1                                                  4            
2                                                  5            
3                                                  4            
4                                                  1            

   Rank Use on Instagram in Terms of Frequency - Feed  \
0                                                  1    
1                                                  1    
2                                                  1    
3                                                  5    
4                                                  2    

   Rank Use on Instagram in Terms of Frequency - Marketplace  \
0                                                  6           
1                                                  6           
2                                                  6           
3                                                  6           
4                                                  3           

   Rank Use on Instagram in Terms of Frequency - Messaging  \
0                                                  3         
1                                                  5         
2                                                  2         
3                                                  2         
4                                                  5         

   Rank Use on Instagram in Terms of Frequency - Create content (post images/stories/reels)  \
0                                                  5                                          
1                                                  3                                          
2                                                  4                                          
3                                                  3                                          
4                                                  6                                          

   Rank Use on Instagram in Terms of Frequency - Stories  \
0                                                  4       
1                                                  2       
2                                                  3       
3                                                  1       
4                                                  4       

  Intentionally Used the Instagram Create Button Before  \
0                                                Yes      
1                                                Yes      
2                                                Yes      
3                                                 No      
4                                                 No      

  Intentionally Used the Instagram Marketplace Button Before  \
0                                                 No           
1     

$H_0$: $\mu_A =\mu_B$ Changing the location of the Create and Marketplace buttons does not have a negative impact on user experience.

$H_A$: $\mu_A \neq \mu_B$ Changing the location of the Create and Marketplace buttons does have a negative impact on user experience.

In [4]:
scipy.stats.levene(data[data.Interface==1]["Rank Placement of Create and Marketplace Button Based on Where Expected"], data[data.Interface==2]["Rank Placement of Create and Marketplace Button Based on Where Expected"])

LeveneResult(statistic=0.4160583941605842, pvalue=0.5227859200159899)

In [5]:
scipy.stats.shapiro(data[data.Interface==1]["Rank Placement of Create and Marketplace Button Based on Where Expected"])

ShapiroResult(statistic=0.8281751871109009, pvalue=0.0023464150726795197)

In [6]:
scipy.stats.shapiro(data[data.Interface==2]["Rank Placement of Create and Marketplace Button Based on Where Expected"]) 

ShapiroResult(statistic=0.8872185945510864, pvalue=0.023913856595754623)

In [7]:
scipy.stats.mannwhitneyu(data[data.Interface==1]["Rank Placement of Create and Marketplace Button Based on Where Expected"], data[data.Interface==2]["Rank Placement of Create and Marketplace Button Based on Where Expected"])

MannwhitneyuResult(statistic=221.0, pvalue=0.5533587259797331)

In [8]:
print("Mean for Interface 1: ", data[data.Interface==1]["Rank Placement of Create and Marketplace Button Based on Where Expected"].mean())
print("Median for Interface 1: ", data[data.Interface==1]["Rank Placement of Create and Marketplace Button Based on Where Expected"].median())
print("Standard error for Interface 1: ", scipy.stats.sem(data[data.Interface==1]["Rank Placement of Create and Marketplace Button Based on Where Expected"]))

print("Mean for Interface 2: ", data[data.Interface==2]["Rank Placement of Create and Marketplace Button Based on Where Expected"].mean())
print("Median for Interface 2: ", data[data.Interface==2]["Rank Placement of Create and Marketplace Button Based on Where Expected"].median())
print("Standard error for Interface 2: ", scipy.stats.sem(data[data.Interface==2]["Rank Placement of Create and Marketplace Button Based on Where Expected"]))

Mean for Interface 1:  2.7
Median for Interface 1:  3.0
Standard error for Interface 1:  0.2064741604835056
Mean for Interface 2:  2.55
Median for Interface 2:  3.0
Standard error for Interface 2:  0.21119958133929795


In [9]:
alt.Chart(data, title="Ranked Placement of Create and Marketplace Button by Interface").mark_bar().encode(
    x=alt.X("Rank Placement of Create and Marketplace Button Based on Where Expected", bin=True, title="Ranked Placement Based on Where Expected"),
    y=alt.Y("count()", title="Count of Each Ranking"),
    color='Interface:N'
)

alt.Chart(...)

Upon an examination of the dependent variable, it was found that the assumption of normality was violated (p < 0.05) using the Shapiro-Wilk test. Therefore,  since we had a between-subjects design with two conditions, a Mann-Whitney test was conducted to compare participant ranking of the placement of the Create and Marketplace buttons on Instagram based on expected locations for Interface 1 and 2. The median placement ranking for Interface 1 (Median = 3.000, Mean = 2.700, SE = 0.207) is the same as that of Interface 2 (Median = 3.000, Mean = 2.550, SE = 0.211) and while the mean placement ranking for Interface 1 is slightly greater than that of Interface 2, the effect size is small. The results of the Mann-Whitney test (W=221, p = 0.553) indicate there is not a statistically significant difference between the rankings of the two interfaces (p > 0.05), meaning we fail to reject the null hypothesis and swapping the location of the two buttons does not have a negative impact on user experience. 